src/DataTypes/Serializations/SerializationUUID.cpp

```diff
@@ -44,10 +45,11 @@ void SerializationUUID::serializeTextQuoted(const IColumn & column, size_t row_n
 
 void SerializationUUID::deserializeTextQuoted(IColumn & column, ReadBuffer & istr, const FormatSettings &) const
 {
+    std::string quoted_string;
```

Comment:
Extra allocation.

Reply:
Yes, it needs somewhere to place the unquoted string. That's assuming we want to deal with quotes (which I thought necessary), otherwise we could look for the next quote, assume that's the end and try to parse the UUID from quote to quote.